src/pytypes/onefuzztypes/models.py

```diff
@@ -799,6 +799,11 @@ class NetworkConfig(BaseModel):
     subnet: str = Field(default="10.0.0.0/16")
 
 
+class NetworkSecurityGroupConfig(BaseModel):
+    allow_service_tags: Optional[List[str]]
```

Comment:
If you choose to apply my suggestion below, then this should become:
```suggestion
    allowed_service_tags: Optional[List[str]]
```
(they should match, whatever you go with)